app/Services/Forge/Pipeline/AnnounceSiteOnSlack.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of Laravel Harbor.
+ *
+ * (c) Mehran Rasulian <mehran.rasulian@gmail.com>
+ *
+ *  For the full copyright and license information, please view the LICENSE
+ *  file that was distributed with this source code.
+ */
+
+namespace App\Services\Forge\Pipeline;
+
+use App\Notifications\SiteProvisionedNotification;
+use App\Services\Forge\ForgeService;
+use App\Traits\Outputifier;
+use Closure;
+use Illuminate\Notifications\Slack\SlackChannel;
+use Illuminate\Notifications\Slack\SlackRoute;
+use Illuminate\Support\Facades\Notification;
+
+class AnnounceSiteOnSlack
+{
+    use Outputifier;
+
+    public function __invoke(ForgeService $service, Closure $next)
+    {
+        // End early if the slack bot token and channel are not set in the Forge service settings
+        if ( ! $service->setting->slackAnnouncementEnabled || ! $service->setting->slackBotUserOauthToken || ! $service->setting->slackChannel ) {
```

Comment:
Do we need to check for the slack token and channel? As we are validating them in the ForgeSetting

Reply:
I tested it out and looks like the validator handles it 🚀. I would resolve the conversation

```bash
  FAIL    ---> The slack bot user oauth token field is required.
  FAIL    ---> The slack channel field is required.
```